### ISSUE TITLE
Overlays 

### ISSUE THREAD

**Issue by chairbender** [unitystation/unitystation#2314]
# Description
Depends on #2311 as it involves working with sprites / animations.

This is used a lot throughout tg - search add_overlay. Create a way to add overlays to items / objects / tiles, such that a particular sprite is overlaid on top of a given object. For example, this would be used to add a stamp to a piece of paper. The API for doing this should be made to be easy to use, and please avoid using strings or enums to reference particular overlays (SOs would be a better alternative most likely). Any commonly used overlays can be referenced in a singleton SO just like how CommonTraits works.

This also needs to be possible to use on tiles, not just for individual objects / items.

This involves networking so the overlays are synced to other players.

Various uses for overlays:

- Adding blood splatter to items
- Scorch marks on the floor
- Blood on the floor / bloody footprints
- wet floor
- stamps on paper

The system should strive to be usable by developers and it should support the same use cases that add_overlay supports in TG.

**Comment by StrawberryMoses** [comment]
This is also used for objects like stamps.
Keep in mind that the overlay will also need to be networked

**Comment by chairbender** [comment]
This could be a good bounty, I think it needs to be fleshed out a bit more.
Look into what capabilities are needed based on what's in TG to ensure we don't miss important functionality. Also look into how overlay assets are set up in TG so our system will be able to re-use those same assets appropriately.

**Comment by chairbender** [comment]
updated description to make it more clear

**Comment by chairbender** [comment]
This will be complicated, so adding a bounty of 100.

**Comment by PerfectTangent** [comment]
there is no longer a bounty associated with this issue

**Comment by PerfectTangent** [comment]
@NoooneyDude didn't you add an overlay system some time recently?

**Comment by NoooneyDude** [comment]
Nope. Currently we can set up overlays for gameobjects by adding a new gameobject in the prefab's hierarchy for that overlay, with its own SpriteHandler. It works and is alright, but isn't as convenient as what is described here.

As for tiles, tile overlays have plenty of room for improvement.
